### DISCUSSION TITLE
Building share packages prior to app build

### DISCUSSION THREAD

**Discussion by dev-tarun-nw** [vercel/turborepo#8990] [Help] [answered]
Hi , 
I want some suggestions related to **turbo repo** and its build process 
**Context** :  currently i have some apps and packages, these some app dependence of internal packages 
**Doubts** : 
*How should i approach the build process for app ?*
* should i do separate build for packages first (to leverage caching optimization during build), before any app
* should i do build package along with the app which are using them.

**My thoughts**
* i think if i build package along with app(which means app is building the package) will makes no difference as a app sharing common dir for packages, but if some how able to build the package separately and cosume them in app it will give benefits like : **less build time for individual app as package is build and cached before any app build**.

**Comment by anthonyshew** [comment, marked as answer]
Trade-offs discussed here!: https://turbo.build/repo/docs/core-concepts/internal-packages#compilation-strategies
